lib/rspec/core/metadata_filter.rb

```diff
@@ -15,22 +15,19 @@ def apply?(predicate, filters, metadata)
         # @private
         def filter_applies?(key, value, metadata)
           silence_metadata_example_group_deprecations do
-            return filter_applies_to_any_value?(key, value, metadata) if Array === metadata[key] && !(Proc === value)
             return location_filter_applies?(value, metadata)          if key == :locations
             return id_filter_applies?(value, metadata)                if key == :ids
             return filters_apply?(key, value, metadata)               if Hash === value
 
             return false unless metadata.key?(key)
+            return true if value.is_a?(TrueClass) && !!metadata[key]
```

Comment:
Since there are so many values in ruby that are truthy (everything but `false` and `nil`!) I'm concerned about this causing a regression for some users.  An example case I can think of is this: the user is using a `Proc` (which is considered truthy) for their filtering.  Previously, it would have hit the `when Proc` branch below and evaulated the proc to determine if the filter applies.  Here you short circuit that...when in fact, the proc may evaluate to `false`.  

There may be other cases that are potential regressions but that's the only one I can think of.  We should figure out a solution for that.